webapp/components/LoginForm/LoginForm.spec.js

```diff
@@ -57,5 +57,13 @@ describe('LoginForm', () => {
         })
       })
     })
+    
+    describe('Click event', () => {
+      it('clicking icon shows/hides password and changes icon', () => {   
+        wrapper.find('a.click-wrapper').trigger('click')
+        expect(wrapper.data.showPassword).toEqual(!showPassword)
+      })
```

Comment:
The objective is, to test what is happening in the browser, and not what is in the vue component. If we check only the value of any data of the component (`wrapper.vm.showPassword`) we do not know what is happening in the browser.
I suggest something like:
```javascript
    describe('click on show password', () => {
      const wrapper = Wrapper()
      it('does not show the password by default', () => {
        expect(wrapper.find('input[name="password"]').attributes('type')).toEqual('password')
      })

      it('shows the password after click on show password', async () => {   
        wrapper.find('a.click-wrapper').trigger('click')
        await Vue.nextTick()
        await expect(wrapper.find('input[name="password"]').attributes('type')).toEqual('text')
      })
    })
```
To get this working you need to `import Vue from 'vue'`. We have to wait a tick until the component applies the changes. Something similar for the change of the icon.
Use `console.log(wrapper.find(...).html())` to see what the rendered component (or a part of it) looks like.